src/Extensions/IAsyncEnumerableExtensions.cs

```diff
@@ -2265,7 +2265,95 @@ private static async Task _enumerate(AsyncEnumerator<TSource>.Yield yield, Disti
 
             return false;
         }
-        
+
+#endregion
+
+#region Count
+        /// <summary>
+        /// Returns the number of elements in a sequence.
+        /// </summary>
+        /// <param name="source">An <see cref="IAsyncEnumerable{T}"/> that contains the elements to apply the predicate to.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the async operation.</param>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <returns>The number of elements in the input sequence.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> or <paramref name="predicate"/> is null.</exception>
+        public static async Task<int> CountAsync<TSource>(
+            this IAsyncEnumerable<TSource> source,
+            CancellationToken cancellationToken = default)
+        {
+            if (source == null)
+                throw new ArgumentNullException(nameof(source));
+
+            var enumerator = source.GetAsyncEnumerator(cancellationToken);
+
+            int count = 0;
+
+            try
+            {
+                while (true)
```

Comment:
[nit] could simplify to while (await enumerator.MoveNextAsync().ConfigureAwait(false))

Reply:
Sure! I was copying the styling from some of the other methods.